src/Autofac/Core/Lifetime/LifetimeScope.cs

```diff
@@ -189,6 +193,108 @@ public ILifetimeScope BeginLifetimeScope(Action<ContainerBuilder> configurationA
+    /// </param>
+    /// <param name="configurationAction">
+    /// Action on a <see cref="ContainerBuilder"/>
+    /// that adds component registrations visible only in the new scope.
+    /// </param>
+    /// <returns>A new lifetime scope.</returns>
+    /// <example>
+    /// <code>
+    /// IContainer cr = // ...
+    /// AssemblyLoadContext pluginContext = // ...
+    /// using (var lifetime = cr.BeginLoadContextLifetimeScope(pluginContext, builder =&gt; {
+    ///         var assembly = pluginContext.LoadFromAssemblyPath("Plugins/plugin.dll");
+    ///         builder.RegisterType(assembly.GetType("PluginEntryPoint")).As&lt;IPlugin&gt;();
+    /// {
+    ///     var plugin = lifetime.Resolve&lt;IPlugin&gt;();
+    /// }
+    /// </code>
+    /// </example>
+    /// <remarks>
+    /// When the returned lifetime scope is disposed, the provided
+    /// <paramref name="loadContext"/> *may* be unloaded, in that
+    /// Autofac will no longer have any references to types loaded from <paramref name="loadContext"/>.
+    /// However if you have captured references to types from the loaded assemblies manually, you still may not be able
+    /// to unload.
+    /// </remarks>
+    public ILifetimeScope BeginLoadContextLifetimeScope(AssemblyLoadContext loadContext, Action<ContainerBuilder> configurationAction)
+    {
+        return BeginLoadContextLifetimeScope(MakeAnonymousTag(), loadContext, configurationAction);
+    }
+
+    /// <summary>
+    /// Begin a new tagged sub-scope, with additional components available to it that may be dynamically
+    /// loaded from the provided <see cref="AssemblyLoadContext"/>.
+    /// Component instances created via the new scope
+    /// will be disposed along with it.
+    /// </summary>
+    /// <param name="tag">The tag applied to the <see cref="ILifetimeScope"/>.</param>
+    /// <param name="loadContext">
+    /// A <see cref="AssemblyLoadContext"/> to associate to the created <see cref="ILifetimeScope"/>.
+    /// </param>
+    /// <param name="configurationAction">
+    /// Action on a <see cref="ContainerBuilder"/>
+    /// that adds component registrations visible only in the new scope.
+    /// </param>
+    /// <returns>A new lifetime scope.</returns>
+    /// <example>
+    /// <code>
+    /// IContainer cr = // ...
+    /// AssemblyLoadContext pluginContext = // ...
+    /// using (var lifetime = cr.BeginLoadContextLifetimeScope(pluginContext, builder =&gt; {
+    ///         var assembly = pluginContext.LoadFromAssemblyPath("Plugins/plugin.dll");
+    ///         builder.RegisterType(assembly.GetType("PluginEntryPoint")).As&lt;IPlugin&gt;();
+    /// {
+    ///     var plugin = lifetime.Resolve&lt;IPlugin&gt;();
+    /// }
+    /// </code>
+    /// </example>
+    /// <remarks>
+    /// When the returned lifetime scope is disposed, the provided
+    /// <paramref name="loadContext"/> *may* be unloaded, in that
```

Comment:
```suggestion
    /// <paramref name="loadContext"/> is allowed to be unloaded, in that
```

Prior phrasing with "may" read like _Autofac_ might be the thing doing the unloading, and that it was non-deterministic. Like, "you know, Autofac may or may not unload this thing."

Reply:
I was seeking wording that indicated that you can unload it, provided you have followed all the other rules around load context unloading.  But it's fair that it sounds like Autofac might not do things correctly.